components/calendar/src/chinese_based.rs

```diff
@@ -252,19 +257,32 @@ impl<C: ChineseBased + CalendarArithmetic> ChineseBasedDateInner<C> {
             "Day should be in range of u8! Value {month_i64} failed for RD {date:?}"
         );
         let day = day_i64 as u8;
-        C::new_chinese_based_date(year, month, day)
+        let is_leap_year = Self::new_year_is_leap_year(first_day_of_year, None);
```

Comment:
Suggestion: I think you can pass in the pre-computed solstice here based on the return value of `new_year_on_or_before_fixed_date` earlier in this function.

Reply:
But that's what you have, right?